jctools-core/src/main/java/org/jctools/queues/MpmcUnboundedXaddArrayQueue.java

```diff
@@ -658,7 +668,96 @@ public boolean relaxedOffer(E e)
     @Override
     public E relaxedPoll()
     {
-        return poll();
+        final int chunkMask = this.chunkMask;
+        final int chunkShift = this.chunkShift;
+        final int chunkSize = chunkMask + 1;
+        final long consumerIndex = this.lvConsumerIndex();
+        final AtomicChunk<E> consumerBuffer = this.lvConsumerBuffer();
+        final int consumerOffset = (int) (consumerIndex & chunkMask);
+        final long chunkIndex = consumerIndex >> chunkShift;
+        final boolean firstElementOfNewChunk = consumerOffset == 0 && consumerIndex >= chunkSize;
+        if (firstElementOfNewChunk)
```

Comment:
Surely this is rare and can be extracted to another method?

Reply:
This is rare, but it always happen, so I wasn't sure to pack it in a separate method: if you think it would improve readability I will do it :+1: